src/pkg/oci/common.go

```diff
@@ -92,38 +155,13 @@ func (o *OrasRemote) withAuthClient(ref registry.Reference) (*auth.Client, error
 	return client, nil
 }
 
-// NewOrasRemote returns an oras remote repository client and context for the given reference.
-func NewOrasRemote(ref registry.Reference) (*OrasRemote, error) {
-	o := &OrasRemote{}
-	o.Context = context.TODO()
-	// patch docker.io to registry-1.docker.io
-	// this allows end users to use docker.io as an alias for registry-1.docker.io
-	if ref.Registry == "docker.io" {
-		ref.Registry = "registry-1.docker.io"
-	}
-	repo, err := remote.NewRepository(ref.String())
-	if err != nil {
-		return &OrasRemote{}, err
-	}
-	repo.PlainHTTP = zarfconfig.CommonOptions.Insecure
-	authClient, err := o.withAuthClient(ref)
+// CheckAuth checks if the user is authenticated to the remote registry.
+func (o *OrasRemote) CheckAuth() error {
+	reg, err := remote.NewRegistry(o.Repository.Reference.Registry)
 	if err != nil {
-		return &OrasRemote{}, err
+		return err
 	}
-	repo.Client = authClient
-	o.Repository = repo
-	return o, nil
-}
-
-// PrintLayerExists prints a success message to the console when a layer has been successfully published to a registry.
-func PrintLayerExists(_ context.Context, desc ocispec.Descriptor) error {
-	title := desc.Annotations[ocispec.AnnotationTitle]
-	var format string
-	if title != "" {
-		format = fmt.Sprintf("%s %s", desc.Digest.Encoded()[:12], First30last30(title))
-	} else {
-		format = fmt.Sprintf("%s [%s]", desc.Digest.Encoded()[:12], desc.MediaType)
-	}
-	message.Successf(format)
-	return nil
+	reg.PlainHTTP = zarfconfig.CommonOptions.Insecure
+	reg.Client = o.client
+	return reg.Ping(o.Context)
```

Comment:
We might want to spin off an issue for this function looking for both push and pull scopes since we don't need both of them in all cases.

Reply:
https://github.com/defenseunicorns/zarf/issues/1821